src/openms/source/ANALYSIS/ID/ConsensusIDAlgorithmSimilarity.cpp

```diff
@@ -94,36 +91,33 @@ namespace OpenMS
         // similarity scores and PEPs of best matches for all ID runs:
         vector<pair<double, double> > best_matches;
         best_matches.reserve(ids.size() - 1);
-        for (vector<PeptideIdentification>::iterator id2 = ids.begin();
-             id2 != ids.end(); ++id2)
+        for (auto& id2 : ids)
         {
-          if (id1 == id2) continue;
+          if (std::addressof(id1) == std::addressof(id2)) continue; //The original iterator-based for loop compared id1 to id2, where id1 and id2 were iterators. 
```

Comment:
please undo the conversion here. this code can easily lead to errors (e.g., just imagine someone removes the & in the range for loop)